http://sourceware.org/ml/gdb-cvs/2011-12/msg00201.html

### src/gdb/ChangeLog	2011/12/19 22:20:05	1.13649
### src/gdb/ChangeLog	2011/12/20 04:29:33	1.13650
## -1,3 +1,9 @@
+2011-12-19  Doug Evans  <dje@google.com>
+
+	* python/py-auto-load.c (info_auto_load_scripts): Pass address of
+	scripts vector to collect_matching_scripts.
+	(collect_matching_scripts): Update.
+
 2011-12-19  Jan Kratochvil  <jan.kratochvil@redhat.com>
 
 	* symfile.c (reread_symbols): Move free_objfile_separate_debug,
--- src/gdb/python/py-auto-load.c	2011/12/10 22:51:47	1.15
+++ src/gdb/python/py-auto-load.c	2011/12/20 04:29:35	1.16
@@ -481,10 +481,10 @@
 collect_matching_scripts (void **slot, void *info)
 {
   struct loaded_script *script = *slot;
-  VEC (loaded_script_ptr) *scripts = info;
+  VEC (loaded_script_ptr) **scripts_ptr = info;
 
   if (re_exec (script->name))
-    VEC_safe_push (loaded_script_ptr, scripts, script);
+    VEC_safe_push (loaded_script_ptr, *scripts_ptr, script);
 
   return 1;
 }
@@ -563,8 +563,9 @@
   if (pspace_info != NULL && pspace_info->loaded_scripts != NULL)
     {
       immediate_quit++;
+      /* Pass a pointer to scripts as VEC_safe_push can realloc space.  */
       htab_traverse_noresize (pspace_info->loaded_scripts,
-			      collect_matching_scripts, scripts);
+			      collect_matching_scripts, &scripts);
       immediate_quit--;
     }
 
